target/linux/realtek/dts/rtl9302_zyxel_xgs1210-12.dts

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/dts-v1/;
+
+#include "rtl930x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "zyxel,xgs1210-12", "realtek,rtl838x-soc";
+	model = "Zyxel XGS1210-12 Switch";
+
+	// mw.l 0xb8003308 0x00200000 active low toggles PHY reset
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0xe0000>;
+				read-only;
+			};
+			partition@e0000 {
+				label = "u-boot-env";
+				reg = <0xe0000 0x10000>;
+				read-only;
+			};
+			partition@f0000 {
+				label = "u-boot-env2";
+				reg = <0xf0000 0x10000>;
+				read-only;
+			};
+			partition@100000 {
+				label = "jffs";
+				reg = <0x100000 0x100000>;
+			};
+			partition@200000 {
+				label = "jffs2";
+				reg = <0x200000 0x100000>;
+			};
+			partition@300000 {
+				label = "firmware";
+				reg = <0x300000 0xd00000>;
+				compatible = "denx,uimage";
```

Comment:
> Installation to flash was not tested.

Not sure about the reason. But since Bjørns "[use device tree properties for non-standard uimage parsing](https://patchwork.ozlabs.org/project/openwrt/list/?series=225499&state=*)" patches are merged, the image splitting should work with the following changes:

```
compatible = "openwrt,uimage", "denx,uimage";
openwrt,partition-magic = <0x93001210>;
```

Reply:
Flashing was not tested because I was not sure to get the flash back into factory configuration. I also want to discourage people who believe this device will already flawlessly run. In contrast to other Realtek switches, u-boot on this device does not properly initialize and patch the 2.5 GBit PHYs and the 10GBit SerDes', the SDK code suggests without proper patching there might be compatibility issues in particular with Marvel chips. For efficient testing I will need to go back and forth for some more time between the original firmware and owrt. I would be happy if someone else wants to add the compatibility properties and try flashing.

Reply:
I'd be happy if we not merge this device support commit then, and keep it open as a separate PR after the rest has been merged. The normal expection is that stuff which is build by buildbots can be expected to work.

Reply:
I agree, the rest should be good to go i think, I just gave it a spin locally and things appear to be in place